neurovault/apps/statmaps/models.py

```diff
@@ -292,26 +289,26 @@ def create(cls, my_file, my_file_name, my_name, my_desc, my_collection_pk, my_ma
 
         return image
 
-    # Celery task to (re)generate glass brain image and image comparisons
+    # Celery task to generate glass brain image on new/update, transform/pearson on update only
     def save(self):
         file_changed = False
         if self.pk is not None:
             existing = Image.objects.get(pk=self.pk)
             if existing.file != self.file:
                 file_changed = True
-        do_update = True if file_changed or self.pk is None else False
+        do_update = True if file_changed else False
+        new_image = True if self.pk is None else False
 
         super(Image, self).save()
-        if do_update and self.collection and self.collection.private == False:
-            generate_glassbrain_image.apply_async([self.pk])
 
-            imgs = Image.objects.filter(collection__private=False).exclude(pk=self.pk)
-            comp_qs = imgs.exclude(polymorphic_ctype__model__in=['image','atlas']).order_by('id')
-            for comp_img in comp_qs:
-                iargs = sorted([comp_img.pk,self.pk])
-                print "Calculating pearson similarity for images %s and %s" % (iargs[0],iargs[1])
-                save_voxelwise_pearson_similarity.apply_async(iargs)
+        if self.collection and self.collection.private == False:
```

Comment:
also you need to delete the pkl (so it would be regenerated if the object was updated) and all of the comparison values related to this object (they also will be regenerated, but it will take some time - no value is better tha noutdated value)